bulk-user-setup/README.md

```diff
@@ -0,0 +1,43 @@
+The script requires following bash utilities. Please install them depending on your OS.
```

Comment:
for my knowledge. Looking at the complexity of the script, any reason why we didn't consider implementing this in Java?
Would had been simpler to implement and maintain.

Reply:
Script solution is as per agreed design. I don't think it's complex (a bit big though).  Java solution might have it's own challenges like 

- it doesn't fit into business domain of any existing micro-service
- new micro-service might be overkill given frequency of it's usage 

@MSancaktutar would you like to add anything?

Reply:
I also thought of a simple java app, initially. But then thinking all aspects, I found the script solution more optimum because of similar reasons @smathangi mentioned.

Reply:
I'm a bit out of context I haven't read the requirements for this piece of work. 
But I would have guessed that this could be implemented with few simple Java classes one of which directly executable with a main method? I was not suggesting a new microservice of course.
It could be in its own github repo. And could be properly tested. 
In my opinion this is a way too far complex bash script. Especially considering most dev are not bash experts. Maintainability hell.

Reply:
For a work that will be maintained for future changes/improvements/new features, you'd be right @mario-paniccia. But this one had appeared to be a one-off development, which, once stabilised, would live untouched for a long while, like the smaller scripts in ccd-docker/bin folder. Thinking again, you may still be right in longer term depending on any feature works on it - let's hope it won't happen but not a huge deal even when so.

Reply:
Then few open questions come up 
-  Where does that project build?
-  How do you publish jar 
- It forces devops to install Java locally to run the program

In any case if we want to build with a java solution then it should come as new story. The original story / confluence page solutionized with Bash script.

Reply:
I appreciate this might be too late now to change, considering other priorities. Maybe something to keep in mind for the future. 
To answer @smathangi questions, I think as far as it builds locally it's ok. Not even need to setup a pipeline and publishing the jar. 
And I think it's fair to assume that DevOp have JRE installed since Java is the main language used in MoJ. I'm sure they'd know how to execute a jar....

Reply:
Also consider that this solution in bash does not allow you to easily test. This is a risk given we are dealing with setting up users on Prod

Reply:
@mario-paniccia  - Not to change this piece of work because of the points already listed all above, but to recognise the value in essence of your suggestion, how about, for some future development of similar with nature a longer term / more frequent usage/maintenance prospect, initiating a new `ccd-cli` java repository and adding any similar work in it from there onwards? We can design a cli for such tasks and any new task can be implemented as a new command in ccd-cli. I agree, it is indeed a fair point to demand JRE installed in any operator machine if it's not already available. Then, release of jar and other stuff can be sorted out with required efforts which would be worth for a strategy of longer term prospect. Having said these, I would like to resolve this comment, if you don't mind, with a take-away of a vision to initiate a java-based ccd-cli in the future as soon as a good fit work comes.